src/itemlistformaction.cpp

```diff
@@ -506,28 +506,31 @@ bool ItemListFormAction::process_operation(Operation op,
 				bool notify = visible_items[0].first->feedurl() != feed->rssurl();
 				for (const auto& item : visible_items) {
 					item.first->set_unread_nowrite_notify(false, notify);
+					invalidate(&item - &visible_items[0]);
 				}
 			}
 			if (cfg->get_configvalue_as_bool("markfeedread-jumps-to-next-unread")) {
 				process_operation(OP_NEXTUNREAD);
+				invalidate_list();
 			} else { // reposition to first/last item
-				std::string sortorder =
-					cfg->get_configvalue("article-sort-order");
-
-				if (sortorder == "date-desc") {
-					LOG(Level::DEBUG,
-						"ItemListFormAction:: "
-						"reset itempos to last");
-					list.set_position(visible_items.size() - 1);
-				}
-				if (sortorder == "date-asc") {
-					LOG(Level::DEBUG,
-						"ItemListFormAction:: "
-						"reset itempos to first");
-					list.set_position(0);
+				if (cfg->get_configvalue_as_bool("show-read-articles")) {
```

Comment:
Marking the feed as read and moving the cursor are two different hings to me and I find it a lot less surprising to not move the cursor on the user if we don't have to  I can only think of two reasons why it was that way; 1) j means "move to the next article unless it's the last article on the list, then it means move to the next feed if goto-next-feed is enabled which is by default".  So by reposition the cursor A follow by j moves to the next feed.  As I said, I think it's confusing to change the position on the user, and we don't lose any functionality as J (or K) would allow user to change feeds irregardless of their current position.  2) current position might be a hidden row so we move to known good position.  This would no longer be an issue.  I moved the invalidate_list to cfg->get_configvalue_as_bool("markfeedread-jumps-to-next-unread"), so this only comes into play if that key is false.  Hmm... might still need to invalidate_list() if show-read-articles (although I tested it and I could not tell a difference) but the call to invalidate_list() is what ends up causing read articles to be hidden so I can't call it it when show-read-articles is yes as that is what allows read articles to be temporarily retained while the feed is open.

Reply:
Oh, I should've just tested it >_< With this PR, the list is *not* empty after I marked everything read — that's the whole point of the PR :) So of course it makes sense to keep the cursor where it was.

No, you don't need to call `invalidate_list()` in this branch then, or do anything else about this comment — everything's fine here.

Thanks for the explanation!